### PR TITLE
Dict: More words2

### PR DESCRIPTION
**Pull request by sebweb3r**

### CODE REVIEW

**Comment by sebweb3r** [comment]
Piecewise isn't in aspell or cambridge dictionary, however it is used in mathematics: https://en.wikipedia.org/wiki/Piecewise

**Comment by peternewman** [comment]
> Piecewise isn't in aspell or cambridge dictionary, however it is used in mathematics: https://en.wikipedia.org/wiki/Piecewise

Hmm, yeah we don't currently have a dictionary for invalid to valid but unknown.

**Review by peternewman — CHANGES_REQUESTED** [review]
A few comments

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -19950,6 +19958,9 @@ piblisher->publisher
 pich->pitch
 picoseond->picosecond
 picoseonds->picoseconds
+pieceweise->piecewise
+piecewiese->piecewise
+piecwise->piecewise
```
[the remark this answers is not in the document]
Comment (sebweb3r) [reply codespell_lib/data/dictionary.txt:19963]: I added the two word version.

File: codespell_lib/data/dictionary_en-GB_to_en-US.txt [hunk 2]
```diff
@@ -100,6 +100,7 @@ generalise->generalize
 generalised->generalized
 generalises->generalizes
 generalising->generalizing
+grey->gray
```
[the remark this answers is not in the document]
Comment (sebweb3r) [reply codespell_lib/data/dictionary_en-GB_to_en-US.txt]: I deleted it.

[the remark this answers is not in the document]
Comment (peternewman) [reply codespell_lib/data/dictionary_en-GB_to_en-US.txt]: It's okay for now, I was just noting for when we get round to auto-reversing the dictionary.

[the remark this answers is not in the document]
Comment (sebweb3r) [reply codespell_lib/data/dictionary_en-GB_to_en-US.txt]: shall i reintroduce it?
Some further suggestions/problems?

[the remark this answers is not in the document]
Comment (sebweb3r) [reply codespell_lib/data/dictionary_en-GB_to_en-US.txt]: @peternewman push

[the remark this answers is not in the document]
Comment (peternewman) [reply codespell_lib/data/dictionary_en-GB_to_en-US.txt]: Yeah I think so please, it just needs considering in #1468

[the remark this answers is not in the document]
Comment (sebweb3r) [reply codespell_lib/data/dictionary_en-GB_to_en-US.txt:103]: Done in 9698d5b

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -12340,6 +12346,7 @@ famlily->family
 famoust->famous
 fanatism->fanaticism
 fancyness->fanciness
+farction->faction, fraction,
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:12349]: I'd probably favour this:
```suggestion
farction->fraction, faction,
```

Grep.app seems to agree.

Can you also add this (but in rare as it's also a medical term):
`infarction->infraction`

**Review by peternewman — CHANGES_REQUESTED** [review]
A comment

**Review by peternewman — APPROVED** [review]

**Pull request merged**